lib/agent.js

```diff
@@ -434,6 +437,12 @@ Agent.prototype.captureError = function (err, opts, cb) {
     : Date.now() * 1000)
   const trans = this._instrumentation.currTransaction()
   const traceContext = (span || trans || {})._context
+  const req = (opts.request instanceof IncomingMessage
+    ? opts.request
+    : trans && trans.req)
+  const res = (opts.response instanceof ServerResponse
+    ? opts.response
+    : trans && trans.res)
 
```

Comment:
REVIEW NOTE: The `trans.req` and `trans.res` need to be grabbed in the sync part of `captureError` because a possible `trans.end()` in the next tick will remove those references.  One of the existing test cases covers this case.